actor/src/main/scala/org/apache/pekko/io/dns/IdGenerator.scala

```diff
@@ -54,7 +56,6 @@ private[pekko] object IdGenerator {
   /**
    * @return a random sequence of ids for production
    */
-  def random(rand: java.util.Random): IdGenerator = new IdGenerator {
-    override def nextId(): Short = rand.nextInt(Short.MaxValue).toShort
-  }
+  def random(rand: java.util.Random): IdGenerator =
+    () => (rand.nextInt(MaxUnsignedShort) - Short.MinValue).toShort
```

Comment:
Since `Short.MinValue` is negative should this be `+ Short.MinValue`? I'm assuming the intent is to move the value into the permitted range for a short before casting. That said I don't think it will alter the behavior as it will overflow and use negative values when cast to short anyway.

Reply:
good point @brharrington - I'll create a new PR